test/images/image-util.sh

```diff
@@ -117,23 +141,46 @@ push() {
   docker_version_check
   TAG=$(<"${IMAGE}"/VERSION)
   if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+    base_images=$(listBaseImages)
+    # NOTE(claudiub): if the REMOTE_DOCKER_URL var is not set, or it is an empty string, we must skip
+    # pushing the Windows image and including it into the manifest list.
+    if [[ ! -v "REMOTE_DOCKER_URL" || -z "${REMOTE_DOCKER_URL}" ]]; then
+      if [[ ! -z "$(printf "%s\n" $base_images | grep '^windows')" ]]; then
```

Comment:
If (pun intended) you can find a way to less deeply nest these conditions I think it might find it easier to follow. Right now the there is a path with no extra processing (if `REMOTE_DOCKER_URL` isn't set but it doesn't include windows in the base images.

Seems like these 2 `ifs` you're adding are really just one longer condition rather than wanting to separately handle up to 4 logical paths (if/else on each).